onnxruntime/core/providers/nuphar/scripts/model_editor.py

```diff
@@ -566,6 +659,141 @@ def convert_rnn_to_scan(node, out_main_graph):
+
+        return outputs
+
+    batch_size, seq_len, feature_size = 1, 40, 160
+
+    input = generate_random_data((seq_len, batch_size, feature_size))
+    hi0 = generate_random_data((12, 1, 600))
+    ci0 = generate_random_data((12, 1, 600))
+    outputs_with_loop = run_with_ort(input_model, [input, hi0, ci0])
+
+    outputs_with_scan = run_with_ort(output_model, [input, hi0, ci0])
+
+    for output_with_loop, output_with_scan in zip(outputs_with_loop, outputs_with_scan):
+        print("loop and scan run diff: ", np.max(output_with_loop - output_with_scan))
+
+    print("")
+
+def convert_loop_to_scan_model(input_model, output_model):
+    in_mp = onnx.load(input_model)
+    out_mp = onnx.ModelProto()
+    out_mp.CopyFrom(in_mp)
+    out_mp.ir_version = 5 # update ir version to avoid requirement of initializer in graph input
+    ensure_opset(out_mp, 9) # bump up to ONNX opset 9, which is required for Scan
+    out_mp.graph.ClearField('node')
+    cast_node_to_remove = []
+    loop_cond_initializer_to_remove = []
+    loop_cond_const_node_to_remove = []
+    for in_n in in_mp.graph.node:
+        if in_n.op_type == 'Loop':
+            scan_op = convert_loop_to_scan(in_n, out_mp.graph)
+
+            # for debugging to make scan output as model graph output
+            # set_op_output_as_model_output(scan_op, out_mp.graph)
+
+            cast_node = None
+            cond_initializer = None
+            cond_const_node = None
+            for n in in_mp.graph.node:
+                if n.op_type == "Cast" and n.output[0] == in_n.input[1]:
+                    cond_initializers = [initializer for initializer in in_mp.graph.initializer if initializer.name == n.input[0]]
+                    cond_const_nodes = [n_c for n_c in in_mp.graph.node if n_c.op_type == "Constant" and n_c.output[0] == n.input[0]]
+                    if len(cond_initializers) == 1:
+                        # TODO: assert the the initializer raw data is not 0 (False)
+                        cast_node = n
+                        cond_initializer = cond_initializers[0]
+                        break
+                    elif len(cond_const_nodes) == 1:
+                        cast_node = n
+                        cond_const_node = cond_const_nodes[0]
+                        break
+                    else:
+                        print("")
+                    
+            if cast_node:
+                cast_node_to_remove = [*cast_node_to_remove, cast_node]
+                if cond_initializer:
+                    loop_cond_initializer_to_remove = [*loop_cond_initializer_to_remove, cond_initializer]
+                elif cond_const_node:
+                    loop_cond_const_node_to_remove = [*loop_cond_const_node_to_remove, cond_const_node]
+            else:
+                raise RuntimeError("Cannot convert a Loop op to Scan: loop cond should be fixed True. Op name = " + cast_n.name)
```

Comment:
>raise RuntimeError("Cannot convert a Loop op to Scan: loop cond should be fixed True. Op name = " + cast_n.name) [](start = 16, length = 112)

Would it better to just skip such Loop ops, instead of raise error?